src/odemis/model/test/futures_test.py

```diff
@@ -310,28 +337,205 @@ def testPF_get_progress(self):
         self.assertEqual(end, endf)
 
         # "start" the task
-        f.set_running_or_notify_cancel()
+        f.set_running_or_notify_cancel()  # updates start and end
         startf, endf = f.get_progress()
-        self.assertLessEqual(startf, time.time())
-        self.assertAlmostEqual(startf + 1, endf)
-        self.assertAlmostEqual(endf, time.time() + 1, delta=0.1)
+        self.assertLessEqual(startf, time.time())  # start should be a tiny bit before now
+        self.assertAlmostEqual(startf + 1, endf)  # task should take 1s, so expect duration is still 1s
+        self.assertAlmostEqual(endf, time.time() + 1, delta=0.1)  # end should be 1s in the future
         time.sleep(0.1)
 
         # "finish" the task
         f.set_result(None)
         self.assertTrue(f.done())
         startf, endf = f.get_progress()
-        self.assertLessEqual(startf, time.time())
-        self.assertLessEqual(endf, time.time())
+        self.assertLessEqual(startf, time.time())  # start should be in the past now as started future in the past
+        self.assertLessEqual(endf, time.time())  # end should be also in the past now as future finished
 
     def cancel_task(self, future):
+        """Task canceller"""
         self.cancelled += 1
         return True
 
     def on_progress_update(self, future, start, end):
+        """Called whenever some progress on the future is reported. Start and end time are updated."""
         self.start = start
         self.end = end
 
+
+class TestProgressiveBatchFuture(unittest.TestCase):
+    """Test progressive batch future."""
+
+    def test_progress_future(self):
+        """
+        Test progress update of batch future.
+        """
+        self.start = None  # for the caller
+        self.end = None  # for the caller
+        fs = {}
+        f1 = ProgressiveFuture()
+        f1.task_canceller = self.cancel_task
+        f2 = ProgressiveFuture()
+        f2.task_canceller = self.cancel_task
+        fs[f1] = 10  # estimated duration in seconds
+        fs[f2] = 15  # estimated duration in seconds
+        time_creation_batch_future = time.time()
+        batch_future = ProgressiveBatchFuture(fs)
+        batch_future.add_update_callback(self.on_progress_update)
+
+        # check estimated time for batch future is sum of estimated time for sub-futures
+        self.assertEqual(self.end - self.start, fs[f1] + fs[f2])
+
+        # "start" the sub-tasks; the batch task is already started at creation
+        f1.set_running_or_notify_cancel()
+
+        now = time.time()
+        f1.set_progress(end=now + 2)  # update the progress on one sub-future
+        # check that the estimated end time of the batch futures has been updated
+        self.assertTrue(now + fs[f2] + 1.9 <= self.end <= now + fs[f2] + 2.1)
```

Comment:
This is not very readable, I would change it to something like this:
```
expected_end = now + fs[f2] + 2 
self.assertTrue(expected_end - 0.1 <= self.end <= expected_end + 0.1)
```

Reply:
same for the other assertions in this test

Reply:
I agree. I just need to check then all other tests and change it there as well as it is nicer to have the same style everywhere.